index.js

```diff
@@ -75,13 +75,16 @@ class PreloadPlugin {
             // value depends on suffix of filename. Otherwise use the given `as` value.
             let asValue;
             if (!options.as) {
-              asValue = entry.match(/\.css$/) ? 'style' : 'script';
+              if (entry.match(/\.css$/)) asValue = 'style';
+              else if (entry.match(/\.(?:eot|otf|ttf|woff2?)$/)) asValue = 'font';
```

Comment:
> This will add preloading for all font types (for example, Font Awesome provides eot, woff, woff2, ttf & svg), are preload capable browsers clever enough to only fetch one that it will later use?

I think (as you note later on) we should focus on the use-case of preloading WOFF2 fonts with the correct type. Browsers won't dedupe multiple font requests for preload.